src/main/java/com/cburch/logisim/analyze/model/Implicant.java

```diff
@@ -57,190 +57,224 @@ public void remove() {
+      for (var key = minimalKey; key < maximalKey; key++) {
+        if (currentTable.containsKey(key) && currentTable.containsKey(key + 1)) {
+          // we see if we can merge terms
+          for (var termGroup1 : currentTable.get(key).keySet()) {
+            for (var termGroup2 : currentTable.get(key + 1).keySet()) {
+              if (termGroup1.unknowns != termGroup2.unknowns)
+                continue;
+              final var differenceMask = termGroup1.values ^ termGroup2.values;
+              if (oneHotTable.contains(differenceMask)) {
+                final var dontCareMask = termGroup1.unknowns | differenceMask;
+                final var newValue = (termGroup1.values & differenceMask) == 0 ? termGroup1.values : termGroup2.values;
+                final var isDontCareGroup = termGroup1.isDontCare && termGroup2.isDontCare;
+                final var newImplicant = new Implicant(dontCareMask, newValue, isDontCareGroup);
+                final var newImplicantTerms = new HashSet<Implicant>();
+                couldMerge = true;
+                termGroup1.isPrime = termGroup2.isPrime = false;
+                newImplicantTerms.addAll(currentTable.get(key).get(termGroup1));
+                newImplicantTerms.addAll(currentTable.get(key + 1).get(termGroup2));
+                if (newTable.containsKey(key)) {
+                  // see if the new implicant already is in the set
+                  var found = false;
+                  for (final var implicant : newTable.get(key).keySet()) {
+                    found |= (implicant.values == newValue) && (implicant.unknowns == dontCareMask);
+                  }
+                  if (!found)
+                    newTable.get(key).put(newImplicant, newImplicantTerms);
+                } else {
+                  newTable.put(key, new HashMap<>());
+                  newTable.get(key).put(newImplicant, newImplicantTerms);
+                }
+              }
+            }
+          } 
         }
       }
-    }
-    toCover.removeAll(covered);
-
-    // This is an unusual case, but it's possible that the
-    // essential prime implicants don't cover everything.
-    // In that case, greedily pick out prime implicants
-    // that cover the most uncovered rows.
-    // BUG: This algorithm does not always find the correct solution
-    // Fix: We are first making a set that contains primes without the don't care set
-    boolean containsDontCare;
-    final var primesNoDontCare = new HashSet<Implicant>();
-    for (final var implicant : primes) {
-      containsDontCare = false;
-      for (final var term : implicant.getTerms()) {
-        if (table.getOutputEntry(term.getRow(), column).equals(Entry.DONT_CARE))
-          containsDontCare = true;
+      // now we add the primes to the set
+      for (final var key : currentTable.keySet()) {
+        for (final var implicant : currentTable.get(key).keySet()) {
+          if (implicant.isPrime && !implicant.isDontCare)
```

Comment:
Probably too long for one-liner so please add code block brackets

Reply:
done